tasks/update-cnb-dependency/load_resources.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/BurntSushi/toml"
+)
+
+type DependencyMetadata struct {
+	Source struct {
+		Name          string `json:"name"`
+		Type          string `json:"type"`
+		VersionFilter string `json:"version_filter"`
+	} `json:"source"`
+	Version struct {
+		Ref    string `json:"ref"`
+		URL    string `json:"url"`
+		Sha256 string `json:"sha256"`
+	} `json:"version"`
+}
+
+type BuildMetadata struct {
+	TrackerStoryID int    `json:"tracker_story_id"`
+	Version        string `json:"version"`
+	Source         struct {
+		URL    string `json:"url"`
+		Md5    string `json:"md5"`
+		Sha256 string `json:"sha256"`
+	} `json:"source"`
+	Sha256 string `json:"sha256"`
+	URL    string `json:"url"`
+}
+
+type DependencyOrchestratorConfig struct {
+	DeprecatedStacks []string          `yaml:"deprecated_stacks"`
+	V3Stacks         map[string]string `yaml:"v3_stacks"`
+	V3DepIDs         map[string]string `yaml:"v3_dep_ids"`
+	V3DepNames       map[string]string `yaml:"v3_dep_names"`
+	IncludeTiny      []string          `yaml:"include_tiny_in_any_stack"`
+}
+
+type EnvVars struct {
+	DeprecationDate DependencyDeprecationDate
+	VersionLine     string
+	VersionsToKeep  int
+}
+
+func loadResources() (EnvVars, DependencyOrchestratorConfig, BuildpackToml, Dependency, BuildMetadata, error) {
```

Comment:
This is an example of where some more composition could be useful. Like this could return a single config object (for example). `Resources () (UpdateConfig, error)`

```go
type UpdateConfig struct {
  Env: Envars,
  Orchestrator: DependencyOrchestratorConfig,
   etc...
}
```

I think this would make things a bit more managable?